engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/SearchPredicateFactory.java

```diff
@@ -301,6 +301,9 @@ SimpleBooleanPredicateOptionsStep<?> or(PredicateFinalStep firstSearchPredicate,
 	 */
 	KnnPredicateFieldStep knn(int k);
 
+	@Incubating
+	ParameterizedPredicateOptionsStep parameterized();
```

Comment:
that'll look as:
```java
var query = index.query().where(
    f -> f.bool()
        .must(
            f.parameterized().predicate(
                ctx -> ctx.predicate()
                    .match()
                    .field( "some-field" )
                    .matching( ctx.param( "my-query-param", String.class ) )
                    .toPredicate() )
        )
        .must(
            f.parameterized().predicate(
                ctx -> ctx.predicate()
                    .simpleQueryString()
                    .field( "some-field" )
                    .matching( ctx.param( "my-simple-query-match-param", String.class ) )
                    .defaultOperator(
                        ctx.param( "my-simple-query-bool-param", BooleanOperator.class ) )
                    .toPredicate() )
        )
)
    .param( "my-query-param", "match me" )
    .param( "my-simple-query-match-param", "I'm a simple query string" )
    .param( "my-simple-query-bool-param", BooleanOperator.OR )
    .toQuery();
```

WDYT @yrodiere

Reply:
That's about what I had in mind... what about:

1. passing the lambda directly to `parameterized`
2. expecting the lambda to return a `PredicateFinalStep` instead of a `Predicate`, to remove the odd `toPredicate()` and be consistent with e.g. `org.hibernate.search.engine.search.predicate.dsl.GenericBooleanPredicateClausesStep#should(java.util.function.Function<? super org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory,? extends org.hibernate.search.engine.search.predicate.dsl.PredicateFinalStep>)`?

Reply:
ok
ok

makes sense will do 😃 thanks